akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1128,12 +1128,56 @@ trait FlowOps[+Out, +Mat] {
    *
    * '''Cancels when''' downstream cancels
    *
-   * @see [[#mapAsyncUnordered]]
+   * @see [[#mapAsyncUnordered]] and [[#mapAsyncPartitioned]]
    */
   def mapAsync[T](parallelism: Int)(f: Out => Future[T]): Repr[T] =
     if (parallelism == 1) mapAsyncUnordered[T](parallelism = 1)(f) // optimization for parallelism 1
     else via(MapAsync(parallelism, f))
 
+  /**
+   * Transform this stream by partitioning elements based on the provided partitioner as they
+   * pass through this step and then applying a given `Future`-returning function to each element
+   * and its partition key.  The value of the returned future, if successful, will be emitted
+   * downstream.
+   *
+   * The number of Futures running at any given time is bounded by the 'parallelism' and 'perPartition'
+   * values.  The futures may complete in any order, but the results are emitted in the same order
+   * as the corresponding elements were received.
```

Comment:
Something is missing here about buffer sizes, the number of _running_ futures is bounded by `parallelism` and `perPartition` but how many completed futures can be queued up, globally and per partition?